testWhitespaceLinterPerformance is flaky on Windows

Test output:
```
Test Case 'WhitespaceLinterPerformanceTests.testWhitespaceLinterPerformance' started at 2025-02-19 08:11:20.962

C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift-format\Tests\SwiftFormatPerformanceTests\WhitespaceLinterPerformanceTests.swift:49: Test Case 'WhitespaceLinterPerformanceTests.testWhitespaceLinterPerformance' measured [Time, seconds] average: 0.083, relative standard deviation: 230.288%, values: [0.625000, 0.016000, 0.031000, 0.016000, 0.031000, 0.015000, 0.032000, 0.015000, 0.032000, 0.015000], performanceMetricID:org.swift.XCTPerformanceMetric_WallClockTime, maxPercentRelativeStandardDeviation: 10.000%, maxStandardDeviation: 0.100

C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift-format\Tests\SwiftFormatPerformanceTests\WhitespaceLinterPerformanceTests.swift:49: error: WhitespaceLinterPerformanceTests.testWhitespaceLinterPerformance : failed: The relative standard deviation of the measurements is 230.288% which is higher than the max allowed of 10.000%.

Test Case 'WhitespaceLinterPerformanceTests.testWhitespaceLinterPerformance' failed (0.832 seconds)
```

This causes a lot of noise in PR-testing. Failures within the last 36 hours were:
https://ci-external.swift.org/job/swift-PR-windows/36617/
https://ci-external.swift.org/job/swift-PR-windows/36620/
https://ci-external.swift.org/job/swift-PR-windows/36623/
https://ci-external.swift.org/job/swift-PR-windows/36649/
https://ci-external.swift.org/job/swift-PR-windows/36660/
https://ci-external.swift.org/job/swift-PR-windows/36668/
https://ci-external.swift.org/job/swift-PR-windows/36671/

The underlying issue is that Swift.org bots for the mainline Windows toolchain are lacking coverage, because developers have no access to the configuration. LLDB tests are one example for this (see https://github.com/swiftlang/swift/issues/79020). They run in PR-testing, but not in mainline testing and thus we have no good monitoring for flakes.

--- a/Tests/SwiftFormatPerformanceTests/WhitespaceLinterPerformanceTests.swift
+++ b/Tests/SwiftFormatPerformanceTests/WhitespaceLinterPerformanceTests.swift
@@ -5,11 +5,19 @@ import XCTest
 @_spi(Testing) import _SwiftFormatTestSupport
 
 final class WhitespaceLinterPerformanceTests: DiagnosingTestCase {
+  /// When executing in Swift CI, run the block to make sure it doesn't hit any assertions because we don't look at
+  /// performance numbers in CI and CI nodes can have variable performance characteristics if they are not bare-metal.
+  ///
+  /// Anywhere else, run XCTest's `measure` function to measure the performance of the block.
+  private func measureIfNotInCI(_ block: () -> Void) {
+    if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] != nil {
+      block()
+    } else {
+      measure { block() }
+    }
+  }
+
   func testWhitespaceLinterPerformance() throws {
-    #if os(Windows)
-    // https://github.com/swiftlang/swift-format/issues/939
-    throw XCTSkip("This test is flaky on Windows")
-    #endif
     let input = String(
       repeating: """
         import      SomeModule
@@ -50,7 +58,7 @@ final class WhitespaceLinterPerformanceTests: DiagnosingTestCase {
       count: 20
     )
 
-    measure { performWhitespaceLint(input: input, expected: expected) }
+    measureIfNotInCI { performWhitespaceLint(input: input, expected: expected) }
   }
 
   /// Perform whitespace linting by comparing the input text from the user with the expected

